packages/pyright-internal/src/analyzer/typeStubWriter.ts

```diff
@@ -158,13 +159,13 @@ export class TypeStubWriter extends ParseTreeWalker {
     private _trackedImportFrom = new Map<string, TrackedImportFrom>();
     private _accessedImportedSymbols = new Set<string>();
 
-    constructor(private _stubPath: string, private _sourceFile: SourceFile, private _evaluator: TypeEvaluator) {
+    constructor(private _stubPath: Uri, private _sourceFile: SourceFile, private _evaluator: TypeEvaluator) {
         super();
 
         // As a heuristic, we'll include all of the import statements
         // in "__init__.pyi" files even if they're not locally referenced
         // because these are often used as ways to re-export symbols.
-        if (this._stubPath.endsWith('__init__.pyi')) {
+        if (this._stubPath.pathEndsWith('__init__.pyi')) {
```

Comment:
Shouldn't this technically be checking `filename` rather than `pathEndsWith` so file names like `foo__init__.pyi` don't get picked up?

Reply:
Good point.